File: UI/angular11/src/app/dtos/order/work-read-dto.ts
import { CityReadDto } from "../city/city-read-dto"
import { UserReadDto } from "../user/user-read-dto"
import { WorkOfferReadDto } from "../work-offer/work-offer-read-dto"

export class WorkReadDto {
    id: number
    creationDate: Date
    prePaid: number
    postPaid: number
    user: UserReadDto
    demands: string
    isRoot: boolean
    city: CityReadDto
    subOrders: WorkReadDto[]
    workOffers: WorkOfferReadDto[]
}

File: ConstructionQualityControl.Domain/Dtos/Order/WorkReadDto.cs
using System;
using System.Collections.Generic;

namespace ConstructionQualityControl.Domain.Dtos
{
    public class WorkReadDto
    {
        public int Id { get; set; }
        public DateTime CreationDate { get; set; }
        public decimal PrePaid { get; set; }
        public decimal PostPaid { get; set; }
        public UserReadDto User { get; set; }
        public string Demands { get; set; }
        public bool IsRoot { get; set; }
        public CityReadDto City { get; set; }

        public List<WorkReadDto> SubOrders { get; set; } = new List<WorkReadDto>();
    }
}
